### PR TITLE
Switcher

### PR DESCRIPTION
**Pull request by Shelenium**

### CODE REVIEW

**Comment by Shelenium** [comment]
Screen Shot 2020-10-20 at 23 19 49

File: projects/client/src/app/store/requests/requests.actions.ts [hunk 1]
```diff
@@ -20,3 +20,11 @@ export class Remove implements WsActionModel<string> {
     constructor(public data: string) {
     }
 }
+
+export class SkipKey implements WsActionModel<{ key: string, skip: boolean }> {
+    static readonly type = '[Requests] Remove';
```
Comment (diprokon) [inline projects/client/src/app/store/requests/requests.actions.ts]: static readonly type = '[Requests] SkipKey';

**Comment by Shelenium** [comment]
Screen Shot 2020-10-22 at 00 30 21

File: projects/client/src/app/store/requests/requests.actions.ts (hunk 1 above)
Comment (Shelenium) [reply projects/client/src/app/store/requests/requests.actions.ts]: +

File: projects/server/src/proxy-mock/store.ts [hunk 2]
```diff
@@ -58,18 +58,20 @@ class Cache extends EventEmitter {
...
-        this.skipKeys[key] = skip;
-        this.write();
+    skip(keys: string[], skip: boolean) {
+        keys.forEach(key => {
+            if (!this.cache.has(key)) {
+                return;
+            }
+            this.cache.get(key).skip = skip;
+            this.skipKeys[key] = skip;
+            this.write();
```
Comment (diprokon) [inline projects/server/src/proxy-mock/store.ts]: this.write(); have to be outside the 'forEach'

File: projects/client/src/app/store/requests/requests.actions.ts [hunk 3]
```diff
@@ -13,10 +13,26 @@ export class GetAll implements WsActionModel {
...
+
+export class SkipKey implements WsActionModel<{ keys: string[], skip: boolean }> {
+    static readonly type = '[Requests] Skip';
+    readonly action = 'skipKey';
+
+    constructor(public data: { keys: string[], skip: boolean }) {
+    }
+}
+
+export class CheckedKey implements WsActionModel<{ key: string, checked: boolean }> {
```
Comment (diprokon) [inline projects/client/src/app/store/requests/requests.actions.ts]: Not used

File: projects/client/src/app/components/requests-table/requests-table.component.scss [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
 }
+
+.mat-column-checkbox {
+    width: 60px;
+}
+
+.extra-header {
+    font-size: large;
+    font-weight: 200;
+    background-color: #ffdd71;
```
Comment (diprokon) [inline projects/client/src/app/components/requests-table/requests-table.component.scss]: background-color: mat-theme-color(...);

**Review by diprokon — APPROVED** [review]

**Pull request merged**